tools/etcd-dump-logs/main.go

```diff
@@ -166,3 +133,175 @@ func excerpt(str string, pre, suf int) string {
 	}
 	return fmt.Sprintf("%q...%q", str[:pre], str[len(str)-suf:])
 }
+
+type EntryFilter func(e raftpb.Entry) (bool, string)
+
+// The 9 pass functions below takes the raftpb.Entry and return if the entry should be printed and the type of entry,
+// the type of the entry will used in the following print function
+func passConfChange(entry raftpb.Entry) (bool, string) {
+	return entry.Type == raftpb.EntryConfChange, "ConfigChange"
+}
+
+func passInternalRaftRequest(entry raftpb.Entry) (bool, string) {
+	var rr etcdserverpb.InternalRaftRequest
+	return entry.Type == raftpb.EntryNormal && rr.Unmarshal(entry.Data) == nil, "InternalRaftRequest"
+}
+
+func passUnknownNormal(entry raftpb.Entry) (bool, string) {
+	var rr1 etcdserverpb.Request
+	var rr2 etcdserverpb.InternalRaftRequest
+	return (entry.Type == raftpb.EntryNormal) && (rr1.Unmarshal(entry.Data) != nil) && (rr2.Unmarshal(entry.Data) != nil), "UnknownNormal"
+}
+
+func passIRRRange(entry raftpb.Entry) (bool, string) {
+	var rr etcdserverpb.InternalRaftRequest
+	return entry.Type == raftpb.EntryNormal && rr.Unmarshal(entry.Data) == nil && rr.Range != nil, "InternalRaftRequest"
+}
+
+func passIRRPut(entry raftpb.Entry) (bool, string) {
+	var rr etcdserverpb.InternalRaftRequest
+	return entry.Type == raftpb.EntryNormal && rr.Unmarshal(entry.Data) == nil && rr.Put != nil, "InternalRaftRequest"
+}
+
+func passIRRDeleteRange(entry raftpb.Entry) (bool, string) {
+	var rr etcdserverpb.InternalRaftRequest
+	return entry.Type == raftpb.EntryNormal && rr.Unmarshal(entry.Data) == nil && rr.DeleteRange != nil, "InternalRaftRequest"
+}
+
+func passIRRTxn(entry raftpb.Entry) (bool, string) {
+	var rr etcdserverpb.InternalRaftRequest
+	return entry.Type == raftpb.EntryNormal && rr.Unmarshal(entry.Data) == nil && rr.Txn != nil, "InternalRaftRequest"
+}
+
+func passIRRCompaction(entry raftpb.Entry) (bool, string) {
+	var rr etcdserverpb.InternalRaftRequest
+	return entry.Type == raftpb.EntryNormal && rr.Unmarshal(entry.Data) == nil && rr.Compaction != nil, "InternalRaftRequest"
+}
+
+func passIRRLeaseGrant(entry raftpb.Entry) (bool, string) {
+	var rr etcdserverpb.InternalRaftRequest
+	return entry.Type == raftpb.EntryNormal && rr.Unmarshal(entry.Data) == nil && rr.LeaseGrant != nil, "InternalRaftRequest"
+}
+
+func passIRRLeaseRevoke(entry raftpb.Entry) (bool, string) {
+	var rr etcdserverpb.InternalRaftRequest
+	return entry.Type == raftpb.EntryNormal && rr.Unmarshal(entry.Data) == nil && rr.LeaseRevoke != nil, "InternalRaftRequest"
+}
+
```

Comment:
Here I added filters for Compaction, LeaseGrant, and LeaseRevoke. More could be added in future if needed.